learning_observer/learning_observer/lms_integration.py

```diff
@@ -0,0 +1,287 @@
+import json
+import recordclass
+import string
+import aiohttp
+import aiohttp.web
+
+import learning_observer
+import learning_observer.runtime
+import learning_observer.google
+import learning_observer.constants as constants
+import learning_observer.settings as settings
+import pmss
+
+pmss.register_field(
+    name="lms_api",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The Canvas Base API URL",
+    required=True
+)
+
+cache = None
+
+class Endpoint(recordclass.make_dataclass("Endpoint", ["name", "remote_url", "doc", "cleaners", "lms"], defaults=["", None])):
```

Comment:
This should have a docstring for what this is and why it exists. It's hard to review without that. 

One major question on all classes is when something should be a class versus a function versus e.g. a JSON object.

Reply:
I have added a docstring